### PR TITLE
Major epicodr update incl. ARDS, 6MWT, EQ5D for children

### LINKED ISSUE #15: feat request to remove rowwise
just a tip, any grouping such as here with rowwise slows down the computation. in calculate_6ils_total you can also just use var1+var2+var3 etc. instead of sum(). then you dont need a grouping anymore.

_Originally posted by @miljukov-o in https://github.com/nukleus-ecu/epicodr/pull/13#discussion_r1253426267_

### LINKED ISSUE #19: The data_primary_coded_suep() function takes several minutes for computation.
For test purposes I ran primary_coding_suep() on the data set "tsExport_Testdata_SUEP_2022-06-27". This data set contains 214 patients and probably lots of missing values. However when I run data_primary_coded_suep() on the data, the process takes several minutes and it feels like R is hanging. But with enough patience, though, the process eventually finishes and creates the primary coded data. 

Can you confirm this with other SÜP data sets?
Is it possible to optimize the calculations, so it's running faster?

I tested it with epicodr git version 0.16.0 from Fri Mar 31 19:30:52 2023 +0200

### PR DESCRIPTION
**Pull request by nuernbergerc**
# Pull Request

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

- Fixes #15 

- Fixes #19 

Update adds several new primary-coding functions.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally in napkon-ecu

**Test Configuration**:
* Tested OS and OS version:
platform       x86_64-w64-mingw32                                   
version.string R version 4.2.3 (2023-03-15 ucrt)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new code warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules/packages

### CODE REVIEW

**Comment by nuernbergerc** [comment]
@miljukov-o @guenther-k 
Liebe beide, 
ich habe nun einige Updates vorgenommen, die geprüft und freigegeben werden müssten. 
Hinsichtlich ARDS der Hinweis: das war dein Code, Olga, wie aus dem Klingenberg-Antrag, nur etwas umgeformt. 
Bzgl. 6MWT: die Formel ist aus dem Paper von Frau Morbach (https://link.springer.com/article/10.1007/s00392-023-02373-3). Könntet ihr beide oder eine*r von euch bitte den Code prüfen und dann freigeben. Danke! 
Die gh-page würde ich nach Update anpassen. 
Liebe Grüße
Carolin

File: R/primary-coding-suep.R [hunk 1]
```diff
@@ -1575,6 +2354,188 @@ build_pcs_score_suep_df_with_proms <- function(trial_data, pid) {
...
+           # Examination date is exact
+           outcome_date_d.factor = ifelse(!is.na(.data$outcome_date.date), "Exakte Angabe (Befunddatum)", NA),
+           ards_rx = case_when(.data$xray_cons == 1 & .data$xray_cons_side == 3 ~ ards_yes,
+                               .data$xray_infil == 1 & .data$xray_infil_side == 3 ~ ards_yes,
+                               # explicitly exclude ards
+                               (.data$xray_cons != 1 | (.data$xray_cons ==1 & .data$xray_cons_side != 3 & .data$xray_cons_side != 4)) &
+                                 (.data$xray_infil != 1 | (.data$xray_infil ==1 & .data$xray_infil_side != 3 & .data$xray_infil_side != 4))  ~ ards_excl,
+                               TRUE ~ as.character(.data$result.factor)),
+           outcome = ifelse(.data$form == "exray", .data$ards_rx, .data$outcome)) %>%
+    rename(mnpfs0.factor_outcome = .data$mnpfs0.factor) %>%
```
Comment (miljukov-o) [inline R/primary-coding-suep.R:2468]: mnpfs0 hier nicht notwendig

File: R/primary-coding-suep.R [hunk 2]
```diff
@@ -1575,6 +2354,188 @@ build_pcs_score_suep_df_with_proms <- function(trial_data, pid) {
...
+  normal_res <- "d Normalbefund"
+  no_res <- "e Kein Befund"
+  no_info <- "e Keine Informationen verf\u00fcgbar"
+  imaging_no_further <- "j Keine weitere Bildgebung"
+  result_unclear <- "k Befund unklar"
+  
+  # ARDS via imaging
+  lung_diag <- trial_data$fv13 %>%
+    mutate(form_parent = "fv13") %>%
+    select(pid, .data$form_parent, .data$mnpfs0.factor, .data$mnpfs1.factor, .data$gec_ct.factor, .data$gec_xray.factor, .data$gec_lus.factor, .data$lmr.factor)
```
Comment (miljukov-o) [inline R/primary-coding-suep.R:2384]: mnpfs0.factor, mnpfs1.factor, nicht notwendig

File: R/primary-coding-suep.R
Comment (miljukov-o) [inline R/primary-coding-suep.R:1]: ggf. die recoding Funktionen auslagern in separates Skript?

**Review by miljukov-o — APPROVED** [review]
only rough sanity check for the new eq5d, cfs and promis functions

**Pull request merged**